src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToProduceReferenceAssembly.cs

```diff
@@ -18,32 +17,30 @@ public class GivenThatWeWantToProduceReferenceAssembly : SdkTest
         public GivenThatWeWantToProduceReferenceAssembly(ITestOutputHelper log) : base(log)
         {}
 
-        [RequiresMSBuildVersionTheory("16.8.0")]
-        [InlineData("netcoreapp3.1", ".csproj", false)]
-        [InlineData("net6.0", ".fsproj", false)]
-        [InlineData("net5.0", ".csproj", true)]
-        [InlineData("net7.0", ".fsproj", true)]
-#pragma warning disable xUnit1025 // InlineData duplicates
-        [InlineData(ToolsetInfo.CurrentTargetFramework, ".csproj", true)]
-        [InlineData(ToolsetInfo.CurrentTargetFramework, ".fsproj", true)]
-#pragma warning restore xUnit1025 // InlineData duplicates
-        public void It_produces_ref_assembly_for_appropriate_frameworks(string targetFramework, string extension, bool expectedExists)
+        [RequiresMSBuildVersionTheory("17.6.0")] // TODO should be 17.7, once things in sync
+        [InlineData("netcoreapp3.1", ".csproj")]
+        [InlineData("net6.0", ".fsproj")]
+        [InlineData("net5.0", ".csproj")]
+        [InlineData("net7.0", ".fsproj")]
+        [InlineData(ToolsetInfo.CurrentTargetFramework, ".csproj")]
+        [InlineData(ToolsetInfo.CurrentTargetFramework, ".fsproj")]
+        public void It_produces_ref_assembly_for_all_frameworks(string targetFramework, string extension)
```

Comment:
The fixes here address the failures that led to https://github.com/dotnet/msbuild/pull/8131